Expand PVC instead of deleting it (#11132)
kyma-project/kyma: docs/migration-guides/1.22-1.23.md

# Migrate from 1.22 to 1.23

## Increase the Loki volume size
In Kyma 1.23.0, the Loki volume size requested by the persistent volume claim has been increased from 10Gi to 30Gi

## Upgrade process

> During the upgrade process, the pre-upgrade job is patching the volume size so as to increase the size from 10Gi to 30Gi as the retention time for logs has been increased from 1 day to 5 days. However, the upgrade process would succeed if the field `allowVolumeExpansion` is set to `true` for the storage class. Please look [here](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#expanding-persistent-volumes-claims) for more info.

Reviewer: ```suggestion
During the upgrade process, the pre-upgrade job increases the volume size from 10Gi to 30Gi. 
For the upgrade process to be successful, make sure that the field `allowVolumeExpansion` is set to `true` for the storage class. For more information, read the [Kubernetes documentation about Expanding Persistent Volumes Claims](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#expanding-persistent-volumes-claims).
```

> If the flag is not set on the storage class then the size of the volume would not change. Use following command to check the size of the volume

Reviewer: ```suggestion
If the flag is not set on the storage class, the size of the volume would not change. Use following command to check the size of the volume:
```

```
kubectl get persistentvolumeclaims -n kyma-system storage-logging-loki-0
```

> if the size is still the old size follow the steps below and then trigger update

Reviewer: ```suggestion
If the size is still the old size, delete the old data with the following command:
```

```
kubectl delete statefulsets.apps -n kyma-system logging-loki

kubectl delete persistentvolumeclaims -n kyma-system storage-logging-loki-0
```

> > This action will delete the existing data.

Reviewer: ```suggestion
After the old data is deleted, trigger the Kyma update.
```

> If there is custom volume size is set for the loki persistent volume claim, then evaluate if the current size would suffice for the logs for retention time of 5 days.

Reviewer: ```suggestion
Make sure that your custom volume size for the Loki persistent volume claim is sufficient for the new retention time of 5 days.
```

I put a similar hint right at the top, so I'm not sure if this is needed here again. Not sure if we want users to evaluate that right away, or at the end.

# Migrate from 1.22 to 1.23

## Increase the Loki volume size
In Kyma 1.23.0, the default Loki volume size requested by the persistent volume claim has been increased from 10Gi to 30Gi because the retention time for logs has been increased from 1 day to 5 days. 

> If you are using a custom volume size for the Loki persistent volume claim, we recommend that you evaluate whether that volume size is sufficient to retain the logs for 5 days.

## Upgrade process
During the upgrade process, the pre-upgrade job increases the volume size from 10Gi to 30Gi. 
For the upgrade process to be successful, make sure that the field `allowVolumeExpansion` is set to `true` for the storage class. For more information, read the [Kubernetes documentation about Expanding Persistent Volumes Claims](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#expanding-persistent-volumes-claims).

> If the flag is not set on the storage class, the size of the volume would not change. Use following command to check the size of the volume:

Reviewer: ```suggestion
If the flag is not set on the storage class, the size of the volume would not change. Use the following command to check the size of the volume:
```

# Migrate from 1.22 to 1.23

## Increase the Loki volume size

> In Kyma 1.23.0, the size of the persistent volume claim has been increased from 10Gi to 30Gi

Reviewer 2: I'd rephrase it a bit since a PVC doesn't have a size, it requests a volume of a certain size. 

``` suggestion
In Kyma 1.23.0, the Loki volume size requested by the persistent volume claim has been increased from 10Gi to 30Gi
```

Reviewer 2: The same wording ( the PVC size) is used below

## Upgrade process
### Custom Loki volume size

If the persistence volume claim size is different from out of box setting (which is 10Gi): If the size is > 30 Gi please use kyma upgarade command by passing the overrides for the desired persistent volume claim size as shown [here](https://kyma-project.io/docs/cli/#details-kyma-cli-command-usage-examples-upgrade-kyma)

Below is an example for override file:
```yaml
apiVersion: v1
data:
  loki.persistence.size: "30Gi"
kind: ConfigMap
metadata:
  labels:
    component: logging
    installer: overrides
  name: logging-overrides
  namespace: kyma-installer
```

### Out of box Loki volume size

However if the persistence volume claim is out of box setting use following command to expand the volume:

```
kubectl delete statefulsets.apps -n kyma-system logging-loki

kubectl patch persistentvolumeclaims -n kyma-system storage-logging-loki-0 --patch '{"spec": {"resources": {"requests": {"storage": "30Gi"}}}}'
```

> > Volume exapansions are only possible if field `allowVolumeExpansion` is set to true. Please look [here](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#expanding-persistent-volumes-claims) for more info.

Reviewer 2: ```suggestion
Volume expansions are only possible if the field `allowVolumeExpansion` of the corresponding storage class is set to true. Please look [here]
```

Below is an example for override file:
```yaml
apiVersion: v1
data:
  loki.persistence.size: "30Gi"
kind: ConfigMap
metadata:
  labels:
    component: logging
    installer: overrides
  name: logging-overrides
  namespace: kyma-installer
```

If the size is between 10Gi and 30Gi please follow the steps mentioned in scenario below.

### Out of box Loki volume size

However if the persistence volume claim is out of box setting use following command to expand the volume:

```
kubectl delete statefulsets.apps -n kyma-system logging-loki

kubectl patch persistentvolumeclaims -n kyma-system storage-logging-loki-0 --patch '{"spec": {"resources": {"requests": {"storage": "30Gi"}}}}'
```
>Volume expansions are only possible if the field `allowVolumeExpansion` of the corresponding storage class is set to true on the storage class. Please look [here]

> If the above process fails then the persistent volume claim template needs to be deleted. Use following command:

Reviewer: ```suggestion
If the process fails, delete the persistent volume claim template with the following command:
```

Which process - the one under "Out-of-box Loki volume size" or the one under "Custom Loki volume size" ?  Additionally, can we make clear how the users see that the operation wasn't successful and that they must use the following fix?

```
kubectl delete statefulsets.apps -n kyma-system logging-loki

kubectl delete persistentvolumeclaims -n kyma-system storage-logging-loki-0
```

> > This action will delete the existing data.

Reviewer: Can we add the information how the users see that the overall operation was successful? Either here, at the end of the entire document, or after the respective instructions that work if there's no error.

And what is the next step? Line 7 announces that the instructions are a preparation for the upgrade, so can we say that the users can now proceed to do the upgrade to 1.23.0?
